test/aida_web/controllers/session_controller_test.exs

```diff
@@ -303,18 +304,15 @@ defmodule AidaWeb.SessionControllerTest do
   end
 
   describe "send_message" do
-    setup do
-      channel = TestChannel.new()
-
-      [channel: channel]
-    end
-
-    test "sends message to a session", %{conn: conn, bot: bot, channel: channel, session: session} do
-
-      with_mock ChannelProvider, [find_channel: fn _ -> channel end] do
-        session_id = session.id
-        post conn, bot_session_session_path(conn, :send_message, bot.id, session_id, message: "Hi!")
-        assert_received {:send_message, ["Hi!"], ^session_id}
+    test "sends message to a session", %{conn: conn, bot: bot, session: session} do
+      session = Session.get(session.id)
```

Comment:
I'm not sure about this line. I had to include it because the test failed while asserting called Bot.send_message. It was being called with a different argument. The difference is in the assets property of the session map. Before this line: ```assets: []```. After this line: ```assets: #Ecto.Association.NotLoaded<association :assets is not loaded>```

Reply:
It's ok. Maybe a bit ugly, but ok. The problem is that an empty array is not the same that an association, even an empty one, so the `send_message` would've been called with a different struct and therefore would fail the assertion.